generate_parameter_library/cmake/generate_parameter_library.cmake

```diff
@@ -92,6 +92,52 @@ function(generate_parameter_library LIB_NAME YAML_FILE)
   install(DIRECTORY ${LIB_INCLUDE_DIR} DESTINATION include/${LIB_NAME})
 endfunction()
 
+
+function(generate_parameter_module LIB_NAME YAML_FILE)
+  find_program(generate_parameter_library_python_BIN NAMES "generate_parameter_library_python")
+  if(NOT generate_parameter_library_python_BIN)
+    message(FATAL_ERROR "generate_parameter_library_python() variable 'generate_parameter_library_python_BIN' must not be empty")
+  endif()
+
+  # Optional 3rd parameter for the user defined validation header
+  if(${ARGC} EQUAL 3)
+    set(VALIDATE_HEADER_FILENAME ${ARGV2})
+  endif()
+
+  # Set the yaml file parameter to be relative to the current source dir
+  set(YAML_FILE ${CMAKE_CURRENT_SOURCE_DIR}/${YAML_FILE})
+
+  set(LIB_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME})
+  file(MAKE_DIRECTORY ${LIB_INCLUDE_DIR})
+
+  find_package(PythonInterp REQUIRED)
+  execute_process(
+          COMMAND "${PYTHON_EXECUTABLE}" -c "import sys;v = sys.version.split()[0];v = v.split('.');print(f'python{v[0]}.{v[1]}')"
+          OUTPUT_VARIABLE PYTHON_VERSION
+          OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  set(PARAM_HEADER_FILE ${CMAKE_INSTALL_PREFIX}/local/lib/${PYTHON_VERSION}/dist-packages/${PROJECT_NAME}/${LIB_NAME}.py)
```

Comment:
This CMake function creates files in `/usr/local/lib`? Why do we want to do that?

I also don't think `${CMAKE_INSTALL_PREFIX}/local/lib` is the right path if we do know for a fact that this is what we want to do.

Reply:
What we want to do is install the Python package to a place on the `PYTHONPATH`. The way ROS 2 goes about this is to add an entry to the `PYTHONPATH` for every Python package built from source in your workspace. In this context,  `CMAKE_INSTALL_PREFIX` refers to `ros_ws/install/my_python_pkg`.